File: api-csharp/Transformations/Resize.cs
using SixLabors.ImageSharp.Processing;
using SharpImage = SixLabors.ImageSharp.Image;

namespace ImageProcessor.Transformations;

class Resize : ITransformation {

    private uint width;

    private uint height;

    public Resize(uint width, uint height) {
        this.width = width;
        this.height = height;
    }

    public void transform(ref SharpImage image) {
        image.Mutate(x => x.Resize((int)width, (int)height));
    }
}

File: api-typescript/src/Transformations/Resize.ts
import sharp from "sharp";
import { ITransformation } from "./ITransformation";

class Resize implements ITransformation {

    private width: number;
    private height: number;

    constructor(width: number, height: number) {
        this.width = width;
        this.height = height;
    }

    public transform(image: sharp.Sharp): sharp.Sharp {
        return image.resize(this.width, this.height, { fit: 'fill' });
    }
}

export { Resize }